src/main/java/org/hmhb/url/DefaultUrlService.java

```diff
@@ -0,0 +1,48 @@
+package org.hmhb.url;
+
+import javax.annotation.Nonnull;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class DefaultUrlService implements UrlService {
+
+    private final HttpServletRequest request;
+    private final Environment environment;
+
+    @Autowired
+    public DefaultUrlService(
+            @Nonnull HttpServletRequest request,
+            @Nonnull Environment environment
+    ) {
+        this.request = requireNonNull(request, "request cannot be null");
+        this.environment = requireNonNull(environment, "environment cannot be null");
+    }
+
+    @Override
+    public String getUrlPrefix() {
+
+        String configuredUrlPrefix = environment.getProperty("hmhb.url.prefix");
+
+        if (StringUtils.isNotBlank(configuredUrlPrefix)) {
```

Comment:
I added this so you can still test parts of the map without always having to deploy to heroku.

Just add a system param for the hmhb.ur.prefix to your .env file:

````
JAVA_OPTS="-Dhmhb.url.prefix=https://hmhb.herokuapp.com -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005"
````